utility/getPipelineStages.py

```diff
@@ -0,0 +1,135 @@
+import json
+import logging
+import os
+import random
+import string
+import sys
+
+import yaml
+from docopt import docopt
+
+log = logging.getLogger(__name__)
+doc = """
+This script fetches all the tests to be run for a pipeline based on the RHCS version and overrides
+
+    Usage:
+        utility/getPipelineStages.py --rhcephVersion <VER> --tags <tags>
+                  [--overrides <str>]
+                  [--metadata <FILE>]
+
+        utility/getPipelineStages.py (-h | --help)
+
+    Options:
+        -h --help          Shows the command usage
+        -v --rhcephVersion VER     The rhcephVersion for which test stages need to be fetched
+        -t --tags <str>    tags to be used for filtering test scripts
+        -o --overrides <str>       Overrides to be considered for execution
+        -m --metadata FILE         Metadata file full path if it is not present in default pipeline/metadata folder
+"""
+
+
+def load_file(file_name):
+    """Retrieve yaml data content from file."""
+    file_path = os.path.abspath(file_name)
+    with open(file_path, "r") as conf_:
+        content = yaml.safe_load(conf_)
+    return content
+
+
+def generate_random_string(length):
+    """Generate a random alphanumeric string of given length"""
+    return "".join(random.choices(string.ascii_uppercase + string.digits, k=length))
+
+
+def fetch_stages(args):
+    """
+    Fetch test stages to be executed based on the input provided
+    Args:
+        args:
+            Dictionary containing the tier_level, cloud_type and all other necessary override parameters
+            to be used to fetch test stages for execution
+
+            Examples:
+                arguments = {
```

Comment:
This is not an example, its description. Example would be - 
```
fetch_stages(rhcephVersion="14.2.22-105", tags="ibm,tier0", overrides={}, metadata_file="metadata.yaml")
```

Reply:
Done.